transport/src/main/java/io/scalecube/transport/TransportImpl.java

```diff
@@ -105,7 +103,7 @@ public TransportImpl(TransportConfig config) {
   private CompletableFuture<Transport> bind0(ServerBootstrap server, InetAddress listenAddress, int bindPort,
       int finalBindPort) {
 
-    incomingMessagesSubject.subscribeOn(Schedulers.from(bootstrapFactory.getWorkerGroup()));
+    incomingMessagesSubject.subscribeOn(Schedulers.fromExecutor(bootstrapFactory.getWorkerGroup()));
```

Comment:
That will not work at all

Reply:
Remember Flux - just immutable builder, so once you executed something without assaining it to the field, then all changes will be lost

Reply:
Still missing point here